frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -795,18 +815,24 @@ pub mod pallet {
                     contract_id.clone(),
                     staker_reward,
                 ));
+
+                weight_info = T::WeightInfo::claim_staker_with_restake();
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
+            // Withdraw reward funds from the dapps staking pot
+            let reward_imbalance = T::Currency::withdraw(
```

Comment:
What if this fails at this point?
What state do we leave the chain in?

Reply:
The failed call will not affect the chain state, since the call by default is transactional.